betty/extension/webpack/assets/working/webpack.config.js

```diff
@@ -0,0 +1,159 @@
+'use strict'
+
+const { CleanWebpackPlugin } = require('clean-webpack-plugin')
+const CssMinimizerPlugin = require('css-minimizer-webpack-plugin')
+const path = require('path')
+const MiniCssExtractPlugin = require('mini-css-extract-plugin')
+const TerserPlugin = require('terser-webpack-plugin')
+const configuration = require('./webpack.config.json')
+const webpack = require('webpack')
+
+class EntryScriptCollector {
+  apply (compiler) {
+    compiler.hooks.initialize.tap('ChunkCollector', () => {
+      compiler.hooks.thisCompilation.tap(
+        'ChunkCollector',
+        (compilation) => {
+          compilation.hooks.processAssets.tapAsync(
+            {
+              name: 'ChunkCollector',
+              stage: compiler.webpack.Compilation.PROCESS_ASSETS_STAGE_OPTIMIZE_INLINE
+            },
+            (_, callback) => {
+              const extensionRegexp = /\.(js|mjs)(\?|$)/
+              const entryScripts = {}
+              for (const entryName of Object.keys(configuration.entry)) {
+                entryScripts[entryName] = compilation.entrypoints
+                  .get(entryName)
+                  .getFiles()
+                  .filter(entryFile => extensionRegexp.test(entryFile))
+              }
+
+              // @todo Refactor entry-scripts.json from a JSON file
+              // @todo to a simply JS file that checks the current page to see which entrypoints are required,
+              // @todo then imports the set of required files for these entrypoints.
+              // @todo
+              compilation.emitAsset(
+                path.join('js', 'entry-scripts.json'),
+                new webpack.sources.RawSource(JSON.stringify((entryScripts)))
+              )
+              return callback()
+            }
+          )
+        }
+      )
+    })
+  }
+}
+
+const webpackConfiguration = {
+  mode: configuration.debug ? 'development' : 'production',
+  devtool: configuration.debug ? 'eval-source-map' : false,
+  entry: configuration.entry,
+  output: {
+    path: path.resolve(__dirname, 'build'),
+    filename: 'js/[name].js'
+  },
+  optimization: {
+    concatenateModules: true,
+    minimize: !configuration.debug,
+    minimizer: [
+      new CssMinimizerPlugin(),
+      new TerserPlugin({
+        extractComments: false,
+        terserOptions: {
+          output: {
+            comments: false
+          }
+        }
+      })
+    ],
+    splitChunks: {
+      chunks: 'all',
+      cacheGroups: {
+        vendorCss: {
```

Comment:
Document that this assumes that the only CSS our entrypoints share comes from node_modules.